---
title: "Differential Expression"
output: html_document
---

Install packages
```{r setup, include=FALSE}
library(DESeq2)
library(tidyverse)
library(forcats)
library(data.table)
library(fgsea)
library(pheatmap)
library(ggplot2)
library(dplyr)
install.packages("openxlsx")
library(openxlsx)
```

Sample-to-sample distance plot
```{r}
#get data
counts <-  as.matrix(read.csv('results/verse_concat_filtered.csv', row.names = 'gene'))

#create distance plot
heatmap(counts, main = "Sample-to-sample distance plot", xlab = "Samples", ylab = "Genes")
```

Call DESeq
```{r}
#fix counts for DESeq
coldata <- data.frame(samples = colnames(counts), treatment = c(rep('CT', 3), rep('KO', 3)), row.names='samples')
coldata$treatment <- as.factor(coldata$treatment)
coldata$treatment <- relevel(coldata$treatment, ref='CT')

#call DESeq
dds <- DESeqDataSetFromMatrix(countData = counts, colData = coldata, design = ~treatment)
dds <- DESeq(dds)
result <- results(dds, contrast=c('treatment', 'CT', 'KO'))

#clean up results
result <- result[order(result$pvalue),]
result %>% as_tibble(rownames='geneid')

#save the data frame as a CSV file
write.csv(result, file = "DE_result.csv", row.names = TRUE)

#remove the na from results
result <- na.omit(result)
```

Histogram of all DE results
```{r}
#create the histogram
hist(result$log2FoldChange, 
     main = "Distribution of log2FoldChanges",
     xlab = "log2FoldChange",
     ylab = "Frequency",
     col = "skyblue",
     border = "black")
```
```{r}
#add a column for significance status based on padj values
volc_result <- as.data.frame(result)
volc_result$regulation <- ifelse(volc_result$log2FoldChange > 0, "Upregulated", "Downregulated")
volc_result$color <- ifelse(volc_result$padj < 0.05, "lightblue", "black")

top_genes <- head(arrange(volc_result, padj), 10)

volcano_plot <- ggplot(volc_result, aes(x = log2FoldChange, y = -log10(padj), color = color)) +
  geom_point() +
  geom_ribbon(aes(xmin= -Inf, xmax=0), alpha=0.25) +
  scale_color_manual(values = c("lightblue", "black"), guide = FALSE) +
  theme_minimal() +
  labs(title = "Volcano Plot",
       x = "log2FoldChange",
       y = "-log10(padj)",
       color = "Significance") +
  geom_vline(xintercept = 0, linetype = "dashed") +  # Add a vertical line at log2FoldChange = 0
  geom_hline(yintercept = -log10(0.05), linetype = "dashed") + # Add a horizontal line at -log10(padj) = padj_threshold
  geom_text(data = top_genes, aes(label = rownames(top_genes)), vjust = 1, hjust = -0.5, size = 3, color = "darkgrey") +  # Label top genes
  geom_text(aes(x = 1.5, y = 12, label = "Blue: Significant Points"), color = "lightblue", hjust = 0, vjust = -0.5, size = 3) +
  geom_text(aes(x = 1.5, y = 12 - 0.5, label = "Black: Insignificant Points"), color = "black", hjust = 0, vjust = -0.5, size = 3) +
  geom_text(aes(x = -3.5, y = -log10(0.05) - 1, label = "Downregulated"), color = "black", hjust = 0, vjust = -0.5, size = 3) +
  geom_text(aes(x = 2.5, y = -log10(0.05) - 1, label = "Upregulated"), color = "black", hjust = 0, vjust = -0.5, size = 3)  

# Display the volcano plot
print(volcano_plot)
```

```{r}
#get mapping
id2gene <- read_delim('results/id2gene.txt', 
                      col_names = c('geneids', 'genenames'),
                      col_types = cols(
                        geneids = col_character(),
                        genenames = col_character()
                      ))
```

FGSEA-all genes
```{r}
#join and tibble
result_geneid <- result %>% 
  as_tibble(rownames='geneids') %>% 
  left_join(id2gene, by='geneids') %>% 
  arrange(padj)

gene_sets <- fgsea::gmtPathways('/projectnb/bf528/materials/project_1_rnaseq/m2.all.v2023.2.Mm.symbols.gmt')

#prepare the data for FGSEA analysis
gene_names <- result_geneid$genenames
log2_fc <- result_geneid$log2FoldChange

#create a named vector of log2 fold change (ranking metric) using gene IDs as names
stats <- setNames(log2_fc, gene_names)

#run GSEA analysis
fgsea_results <- fgsea(pathways = gene_sets,
                       stats = stats,  # Using log2 fold changes as statistics
                       minSize = 15,
                       maxSize = 500)

fgsea_results %>% arrange(padj)

# Extract interesting results (e.g., top enriched gene sets)
top_results <- fgsea_results[order(fgsea_results$padj), ]
write.xlsx(top_results, "fgsea_table_all.xlsx")
```

FGSEA-Statistically Significant DE Genes
```{r}
#filter the dataframe based on the FDR threshold
result <- result[result$padj < 0.05, ]
num_rows <- nrow(result)
num_rows

#join and tibble
result_geneid <- result %>% 
  as_tibble(rownames='geneids') %>% 
  left_join(id2gene, by='geneids') %>% 
  arrange(padj)

#prepare the data for FGSEA analysis
gene_names <- result_geneid$genenames
log2_fc <- result_geneid$log2FoldChange

#create a named vector of log2 fold change (ranking metric) using gene IDs as names
stats <- setNames(log2_fc, gene_names)

#run GSEA analysis
fgsea_results <- fgsea(pathways = gene_sets,
                       stats = stats,  # Using log2 fold changes as statistics
                       minSize = 15,
                       maxSize = 500)

fgsea_results %>% arrange(padj)

# Extract interesting results (e.g., top enriched gene sets)
top_results_ss <- fgsea_results[order(fgsea_results$padj), ]
write.xlsx(top_results_ss, "fgsea_table_sig.xlsx")
```

Enrichment Analysis
```{r}
#load list of DE genes
gene_list <- (result1$genenames)
for (gene in gene_list) {
  print(gene)
}
```
