NAMES = ["ADrep1","ADrep2","P0rep1","P0rep2","P4rep1","P4rep2","P7rep1","P7rep2"]

rule all:
	input:
		'results/GRCm39.primary_assembly.genome.fa.gz',
		'results/gencode.vM33.primary_assembly.annotation.gtf',
		expand('results/full_data/{names}_flagstats.txt', names=NAMES)

#THIS WGET DOES NOT WORK, USED THE PROVIDED FILE FROM THE MATERIALS DIRECTORY
#
#rule wget_m39:
#	output:
#		'results/GRCm39.primary_assembly.genome.fa.gz'
#	params:
#		ftp_path = 'https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M34/GRCm39.primary_assembly.genome.fa.gz'
#	shell:
#		'''
#		wget -P results/ {params.ftp_path}
#		'''

#THIS WGET ALSO DOES NOT WORK< USED THE PROVIDED FILE FROM THE MATERIALS DIRECTORY
#
#rule wget_m39_gtf:
#output:
#		'results/gencode.vM33.primary_assembly.annotation.gtf.gz'
#	params:
#		ftp_path = 'https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M34/gencode.vM34.primary_assembly.annotation.gtf.gz'
#	shell:
#		'''
#		wget -P results/ {params.ftp_path}
#		'''
		
rule unzip:
	input:
		gz = 'results/gencode.vM33.primary_assembly.annotation.gtf.gz'
	output:
		gtf = 'results/gencode.vM33.primary_assembly.annotation.gtf'
	shell:
		'''
		gunzip -c {input.gz} > {output.gtf}
		'''
		
rule star:
	input:
		r1 = 'samples/full_files/{name}_R1.fastq.gz',
		r2 = 'samples/full_files/{name}_R2.fastq.gz',
		star_dir = 'samples/m39_star/'
	output:	
		bam = 'results/full_data/{name}.Aligned.out.bam'
	params:
		prefix = 'results/full_data/{name}.'
	threads: 8
	shell:
		'''
		STAR --runThreadN {threads} --genomeDir {input.star_dir} --readFilesIn {input.r1} {input.r2} --readFilesCommand zcat --outSAMtype BAM Unsorted --outFileNamePrefix {params.prefix}
		'''

rule samtools_flagstat:
	input:
		bfile= 'results/full_data/{name}.Aligned.out.bam'
	output:
		fs = 'results/full_data/{name}_flagstats.txt'
	shell:
		'''
		samtools flagstats {input.bfile} > {output.fs}
		'''
